### PR TITLE
[dashboard] Re-implement default IDE switch for new dashboard

### PR DESCRIPTION
**Pull request by jankeromnes**
Also fix styling of Account and other Settings pages.

TODOs:
- ~~Reimplement a global Settings commit mechanism to actually change values for the user~~

### CODE REVIEW

**Comment by jankeromnes** [comment]
Werft already called it a day 😂

/werft run

**Comment by jankeromnes** [comment]
Now looks good and has a few other visual improvements.

It's just lacking a Settings commit mechanism to actually change the value on the user.

**Comment by svenefftinge** [comment]
👍 
Some comments:
 - Please remove the third option and the possibility specify an image.
 - No need to tell users that Gitpod uses code as default (there is no default as I have to choose between the two).

**Comment by jankeromnes** [comment]
Thanks!

> Please remove the third option and the possibility specify an image.

I can hide it when `ide-settings` permission is off (as in the old dashboard)

> No need to tell users that Gitpod uses code as default (there is no default as I have to choose between the two).

I agree. @gtsiolis do you have an alternative sub-text in mind? Maybe something like:

> Choose which IDE your Gitpod workspaces should start with.

**Comment by svenefftinge** [comment]
> Choose which IDE your workspaces should use.

It's not the workspace that uses an IDE but the user. So how about:

> Choose which IDE you want to use.

**Review by svenefftinge — APPROVED** [review]

**Comment by jankeromnes** [comment]
Ok, the page now actually changes the user settings in the DB:

```
mysql> select * from d_b_user;
+--------------------------------------+--------------------------+----------------------------------------------------+--------------------------+----------------+------------------------------+---------+----------------------------+------------+---------------+-----------------+--------------+--------------------------+---------------------------------------+
| id                                   | creationDate             | avatarUrl                                          | name                     | fullName       | allowsMarketingCommunication | blocked | _lastModified              | privileged | markedDeleted | noReleasePeriod | featureFlags | rolesOrPermissions       | additionalData                        |
+--------------------------------------+--------------------------+----------------------------------------------------+--------------------------+----------------+------------------------------+---------+----------------------------+------------+---------------+-----------------+--------------+--------------------------+---------------------------------------+
| 2f49d29c-1f44-453f-ae9a-1fe2d3a46d49 | 2021-03-18T16:03:55.544Z | https://avatars.githubusercontent.com/u/599268?v=4 | jankeromnes              | Jan Keromnes   |                            0 |       0 | 2021-03-18 18:21:29.138004 |          0 |             0 |               0 | NULL         | ["admin","ide-settings"] | {"ideSettings":{"defaultIde":"code"}} |
+--------------------------------------+--------------------------+----------------------------------------------------+--------------------------+----------------+------------------------------+---------+----------------------------+------------+---------------+-----------------+--------------+--------------------------+---------------------------------------+
```

However, my new workspaces still all start with Theia for some reason. 🤔

**Comment by svenefftinge** [comment]
Please remove the image form.
Screenshot 2021-03-19 at 08 42 52

Also, please make it so that new users use Code as default.

File: components/dashboard/src/settings/Account.tsx [hunk 1]
```diff
@@ -20,30 +19,30 @@ export default function Account() {
...
                         <img className="rounded-full w-24 h-24 border-2 border-transparent hover:border-indigo-400"
                             src={user!.avatarUrl} alt={user!.name} />
                     </div>
                 </div>
             </div>
-            <h3 className="pt-14">Delete Account</h3>
+            <h3 className="mt-12">Delete Account</h3>
             <p className="text-sm text-gray-400 pb-4">This action will remove all the data associated with your account in Gitpod.</p>
-            <button className="border-red-900 bg-red-500 hover:bg-red-700" onClick={() => setModal(true)}>Delete Account</button>
+            <button className="border-red-600 text-red-600 bg-white hover:border-red-800 hover:text-red-800" onClick={() => setModal(true)}>Delete Account</button>
```
Comment (svenefftinge) [inline components/dashboard/src/settings/Account.tsx:45]: Is that `white-bg` based on a design I missed?

Comment (jankeromnes) [reply components/dashboard/src/settings/Account.tsx:45]: That's a secondary button (outline vs plain color), cc @gtsiolis

Currently, it seems that all `<button>` have `bg-green-600` at the moment:

https://github.com/gitpod-io/gitpod/blob/8ff6ed7368f45b3debf996997b38abaa15ffb29c/components/dashboard/src/index.css#L31-L33

**Comment by jankeromnes** [comment]
> Please remove the image form.

How do you get to see that? Here's what I see:

Screenshot 2021-03-19 at 08 50 45
Screenshot 2021-03-19 at 08 50 52

**Comment by svenefftinge** [comment]
> How do you get to see that? Here's what I see:

I don't know I just see it :-)

But you can really remove al the code around a custom IDE image for now, as we are not sure we want to go down that path.

File: components/dashboard/src/settings/Account.tsx (hunk 1 above)
Comment (gtsiolis) [reply components/dashboard/src/settings/Account.tsx:45]: For more context, see relevant specs in https://github.com/gitpod-io/gitpod/issues/3395.

**Comment by jankeromnes** [comment]
Works well now!

**Pull request merged**